iothub_client/tests/iothubtransport_mqtt_common_int_ut/iothubtransport_mqtt_common_int_ut.c

```diff
@@ -702,13 +702,13 @@ static void CreateUrlDecodedMessageProperties(const TEST_EXPECTED_MESSAGE_PROPER
     {
         size_t numberOfApplicationProperties = expectedMessageProperties->applicationProperties->keysLength;
 
-        expectedUrlDecodedMessageProperties->applicationProperties = calloc(1, sizeof(*expectedUrlDecodedMessageProperties->applicationProperties));
+        expectedUrlDecodedMessageProperties->applicationProperties = (TEST_EXPECTED_APPLICATION_PROPERTIES*)calloc(1, sizeof(*expectedUrlDecodedMessageProperties->applicationProperties));
```

Comment:
By design the c99 spec says the instance should not be cast to any type out of malloc or calloc.
But understood here that it is failing for cpp build, so it should be fine.

Reply:
Same for everywhere else in the PR.